src/Analyzers/Core/Analyzers/UseCompoundAssignment/AbstractUseCompoundAssignmentDiagnosticAnalyzer.cs

```diff
@@ -34,6 +34,7 @@ internal abstract class AbstractUseCompoundAssignmentDiagnosticAnalyzer<
             ISyntaxFacts syntaxFacts,
             ImmutableArray<(TSyntaxKind exprKind, TSyntaxKind assignmentKind, TSyntaxKind tokenKind)> kinds)
             : base(IDEDiagnosticIds.UseCompoundAssignmentDiagnosticId,
+                   EnforceOnBuildValues.UseCompoundAssignment,
```

Comment:
💭 Thinking about this some more, it seems like we might want to make a new type that wraps a diagnostic ID _and_ an `EnforceOnBuild` value in one place, and provide an instance from that type to the constructor. This would make it easy to review future additions and prevent copy/paste errors.

Reply:
I personally prefer `EnforceOnBuildValues` to be separate as that allows one to see each bucket of rules as a group, while `IDEDiagnosticIds` allows a sequential list of diagnostic IDs.